### PR TITLE
[kube-prometheus-stack] Update Thanos Ruler to v0.34.0

### PR DESCRIPTION
**Pull request by dragoangel**
#### What this PR does / why we need it
Updates Thanos Ruler to v0.34.0 with support of `keep_firing_for` and other improvements, changelog available here: https://github.com/thanos-io/thanos/blob/main/CHANGELOG.md#v0340---260124

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

**Review by jkroepke — DISMISSED** [review]
LGTM, please bump minor version, thanks.

File: charts/kube-prometheus-stack/Chart.yaml [hunk 1]
```diff
@@ -23,7 +23,7 @@ name: kube-prometheus-stack
 sources:
   - https://github.com/prometheus-community/helm-charts
   - https://github.com/prometheus-operator/kube-prometheus
-version: 56.3.0
+version: 56.3.1
```
Comment (jkroepke) [inline charts/kube-prometheus-stack/Chart.yaml:26]:
```suggestion
version: 56.4.0
```

Comment (dragoangel) [reply charts/kube-prometheus-stack/Chart.yaml:26]: Yes, looks okay for me

**Comment by dragoangel** [comment]
@jkroepke one note: we also need get updated `prometheus-operator` as current version `v0.71.2` says:
```
level=warn ts=2024-01-31T15:57:45.366590632Z caller=rules.go:136 component=thanos-controller thanos=monitoring-thanos-ruler namespace=monitoring prometheusrule=test-thanos prometheusrule-namespace=test msg="ignoring `keep_firing_for` as it is not yet supported in thanos, see https://github.com/thanos-io/thanos/issues/6165"
```

**Review by jkroepke — APPROVED** [review]

**Comment by dragoangel** [comment]
@jkroepke I missed that we have thanos in 2 places: thanos sidecar, and thanos ruller, will create new PR to fix this*

**Pull request merged**